### PR TITLE
feat(component): add new slots for enhanced customization

### PR DESCRIPTION
**Pull request by ricardschmidt**
## Objective

Enhance the component's flexibility by introducing new slots that cater to specific customization needs, such as custom tips, error handling, and improved menu control.

## Work Done

- Added the following slots with detailed functionalities:
  - `control-icon`: For custom icons in the control area.
  - `control-x`: Delete button with customizable classes via the `xClass` prop.
  - `search-prompt-tip`, `loading-tip`: Customizable tips for asynchronous search and loading states.
  - `no-available-tip`, `no-results-tip`: Customizable messages for unavailable or no results found scenarios.
  - `root-error-tip`, `search-loading-error-tip`: Error handling slots with retry functionalities.
- Enhanced `before-list` and `after-list` slots with `toggleMenu` functionality for better interaction control.
- Tested integration and ensured backward compatibility with existing implementations.

## Screenshots / Output

_No significant UI changes by default. Slots demonstrated in custom implementation scenarios._

## Tested

- Verified each slot's functionality independently and in combination.
- Tested props passing to ensure accurate and consistent behavior.
- Confirmed menu toggle behavior in `before-list` and `after-list` slots works as expected.

### CODE REVIEW

File: docs/App.vue [hunk 1]
```diff
@@ -330,6 +330,18 @@
         - CSS classnames for making the style correct
         <demo name="CustomizeOptionLabel" />
 
+        <section-header name="Customize Value Label" :level="2" />
+
+        You can customize the label of value item (each item in case of
+        multi-select). vue3-treeselect utilizes Vue's scoped slot feature and
+        provides some props you should use in your customized template: -
+        <code>node</code>
+        - a normalized node object (note that, this is differnt from what you
```
Comment (zanmato) [inline docs/App.vue:339]: Typo `different`

File: docs/components/DocSlots.vue [hunk 2]
```diff
@@ -53,13 +53,69 @@ export default {
...
+      },
+      {
+        name: "loading-tip",
+        props: makePropList(["text"]),
+        description: `Slot for custom loading tip`
+      },
+      {
+        name: "no-available-tip",
+        props: makePropList(["text"]),
+        description: `Slot for custom no available tip`
```
Comment (zanmato) [inline docs/components/DocSlots.vue:84]: Add "no available options", I was initially confused what "no available" meant :smile:

File: docs/App.vue (hunk 1 above)
Comment (ricardschmidt) [reply docs/App.vue]: thanks

File: docs/App.vue [hunk 3]
```diff
@@ -422,14 +447,23 @@ const sections = [
...
       {
-        name: "Customize Option Label"
+        name: "Customize Control Icon"
       },
       {
-        name: "Customize Value Label"
+        name: "Customize Control X"
+      },
+      {
+        name: "CustomizeAsyncSearchingTip"
```
[the remark this answers is not in the document]
Comment (ricardschmidt) [reply docs/App.vue]: done

File: docs/components/DocSlots.vue (hunk 2 above)
Comment (ricardschmidt) [reply docs/components/DocSlots.vue]: Great suggestion! I also took the opportunity to update the “no results” as well.

**Pull request merged**